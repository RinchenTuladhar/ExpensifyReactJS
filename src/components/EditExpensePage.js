import React from "react";
import { connect } from "react-redux";
import ExpenseForm from "./ExpenseForm";
import { editExpense, removeExpense } from "../actions/expenses";

// Refactor EditExpensePage to be a class based component

class EditExpensePage extends React.Component {
  onSubmit = (expense) => {
    this.props.dispatch(editExpense(this.props.expense.id, expense));
    this.props.history.push("/");
  };

  onRemove = () => {
    this.props.dispatch(removeExpense({ id: this.props.expense.id }));
  };
  render() {
    return (
      <div>
        <ExpenseForm expense={this.props.expense} onSubmit={this.onSubmit} />
        <button onClick={this.onRemove}>Remove</button>
      </div>
    );
  }
}

const mapStateToProps = (state, props) => ({
  expense: state.expenses.find(
    (expense) => expense.id === props.match.params.id
  ),
});

const mapDispatchToProps = (dispatch) => ({
  editExpense: (id, expense) => dispatch(editExpense(id, expense)),
  removeExpense: (data) => dispatch(removeExpense(data)),
});

export default connect(mapStateToProps)(EditExpensePage);
