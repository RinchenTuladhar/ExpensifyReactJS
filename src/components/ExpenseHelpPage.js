const ExpenseHelpPage = () => {
  return (
    <div>
      <h1>Help Component</h1>
    </div>
  );
};
export default ExpenseHelpPage;
