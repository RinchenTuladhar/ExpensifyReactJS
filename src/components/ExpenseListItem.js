import React from "react";
import { Link } from "react-router-dom";

const ExpenseListItem = ({ id, description, amount, createdAt }) => (
  <div>
    <Link to={"/edit/" + id}>Edit</Link>
    <p>{description}</p>
    <p>
      {amount} - {createdAt}
    </p>
  </div>
);

export default ExpenseListItem;
