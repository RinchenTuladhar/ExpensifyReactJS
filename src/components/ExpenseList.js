import React from "react";
import { connect } from "react-redux";
import ExpenseListItem from "./ExpenseListItem";
import selectExpenses from "../selectors/expenses";
import numeral from "numeral";

export const ExpenseList = (props) => (
  <div>
    <p className="total-calculator">
      You have {props.expenses.length} expenses, totalling{" "}
      {numeral(expensesTotal(props.expenses)).format("$0,0.00")}
    </p>

    {props.expenses.length === 0 ? (
      <p>No expenses</p>
    ) : (
      props.expenses.map((expense) => {
        return <ExpenseListItem key={expense.id} {...expense} />;
      })
    )}
  </div>
);

const expensesTotal = (expenses) => {
  let total = 0;

  if (expenses.length > 0) {
    console.log(
      expenses.map((expense) => {
        total += expense.amount;
      })
    );
  }

  return total / 100;
};

const mapStateToProps = (state) => {
  return {
    expenses: selectExpenses(state.expenses, state.filters),
  };
};

export default connect(mapStateToProps)(ExpenseList);
