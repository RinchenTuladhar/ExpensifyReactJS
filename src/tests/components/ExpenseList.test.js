import React from "react";
import { shallow } from "enzyme";
import { ExpenseList } from "../../components/ExpenseList";
import expenses from "../fixtures/expenses";

test("should render ExpenseList with expenses", () => {
  const wrapper = shallow(<ExpenseList expenses={expenses} />);
  expect(wrapper).toMatchSnapshot();
});

test("should render ExpenseList with empty message", () => {
  const wrapper = shallow(<ExpenseList expenses={[]} />);
  expect(wrapper).toMatchSnapshot();
});

test("should return 0 if no expenses return", () => {
  const wrapper = shallow(<ExpenseList expenses={[]} />);
  expect(wrapper).toMatchSnapshot();
});

test("should correctly add up a single expense", () => {});

test("should correcty add up multiple expense", () => {});
