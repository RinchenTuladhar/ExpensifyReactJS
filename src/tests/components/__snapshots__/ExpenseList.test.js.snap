// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseList with empty message 1`] = `
<div>
  <p>
    No expenses
  </p>
</div>
`;

exports[`should render ExpenseList with expenses 1`] = `
<div>
  <ExpenseListItem
    amount={2500}
    createdAt={0}
    description="Bum bag"
    id="1"
    note=""
  />
  <ExpenseListItem
    amount={5000}
    createdAt={86400000}
    description="Duffle bag"
    id="2"
    note=""
  />
  <ExpenseListItem
    amount={4000}
    createdAt={172800000}
    description="Backpack"
    id="3"
    note=""
  />
</div>
`;
